keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/jackson/SecurityGroupRuleDeserializer.kt

```diff
@@ -5,8 +5,12 @@ import com.netflix.spinnaker.keel.api.ec2.CrossAccountReferenceRule
 import com.netflix.spinnaker.keel.api.ec2.ReferenceRule
 import com.netflix.spinnaker.keel.api.ec2.SecurityGroupRule
 import com.netflix.spinnaker.keel.jackson.PropertyNamePolymorphicDeserializer
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
+import org.springframework.boot.jackson.JsonComponent
 
-internal class SecurityGroupRuleDeserializer :
+@JsonComponent
+@ConditionalOnMissingBean(SecurityGroupRuleDeserializer::class)
```

Comment:
`SecurityGroupRuleDeserializer` is made conditional on any sub-classes such that we can replace it with our own in `keel-nflx`.